downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/Biopax.java

```diff
@@ -0,0 +1,211 @@
+package org.reactome.release.downloadDirectory;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.biopax.validator.api.Validator;
+import org.biopax.validator.api.ValidatorUtils;
+import org.biopax.validator.api.beans.Validation;
+import org.biopax.validator.impl.IdentifierImpl;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.reactome.biopax.SpeciesAllPathwaysConverter;
+import org.reactome.biopax.SpeciesAllPathwaysLevel3Converter;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.Set;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+
+public class Biopax {
+  private static final Logger logger = LogManager.getLogger();
+  static ApplicationContext ctx;
+  static boolean autofix = false;
+  static int maxErrors = 0;
+  static String profile = "notstrict";
+  static String outFormat = "xml";
+  static Validator validator;
+
+  // BioPAX validation requires loading of bio ontologies, which takes a few minutes before starting the BioPAX process
+  static {
+   logger.info("Running BioPAX step");
+   logger.info("Preparing BioPAX validation rules...");
+    ctx = new ClassPathXmlApplicationContext(new String[]{
+      "META-INF/spring/appContext-validator.xml",
+      "META-INF/spring/appContext-loadTimeWeaving.xml"
+    });
+    validator = (Validator) ctx.getBean("validator");
+    logger.info("Finished preparing BioPAX validation rules");
+  }
+  
+  // Once BioPAX validation rules have loaded, the actual BioPAX process can start
+  public static void execute(String username, String password, String host, String port, String database, String releaseNumber, String pathToSpeciesConfig, boolean runBioPAX2, boolean runBioPAX3) throws Exception {
```

Comment:
The method signature is pretty long.  It might be worth creating a private inner class to have an object that holds the username, password, host, port, and database.  That way the database parameters are together and then the object can be passed in as the first parameter and the values extracted in the body.

Reply:
Could the booleans runBioPAX2 and runBioPAX3 be replaced with the list of biopaxLevels (as integers) to run?  That way the levels don't need to be hard-coded in the method body and you don't need to check which level you're running in the loop when calling generateBioPAXFile.

Reply:
I made the change for the second point, but I don't quite understand the first one